health-checker/src/main/java/io/stargate/health/StorageHealthChecker.java

```diff
@@ -0,0 +1,147 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.health;
+
+import com.codahale.metrics.health.HealthCheck;
+import com.datastax.oss.driver.api.core.uuid.Uuids;
+import io.stargate.db.datastore.DataStore;
+import io.stargate.db.datastore.DataStoreFactory;
+import io.stargate.db.datastore.ResultSet;
+import io.stargate.db.datastore.Row;
+import io.stargate.db.query.Predicate;
+import io.stargate.db.query.builder.Replication;
+import io.stargate.db.schema.Column;
+import io.stargate.db.schema.ImmutableColumn;
+import io.stargate.db.schema.ImmutableTable;
+import io.stargate.db.schema.Table;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class StorageHealthChecker extends HealthCheck {
+  private static final Logger logger = LoggerFactory.getLogger(DataStoreHealthChecker.class);
+  private static final boolean SHOULD_CREATE_KS_AND_TABLE =
+      Boolean.parseBoolean(
+          System.getProperty("stargate.health_check.data_store.create_ks_and_table", "true"));
+  private static final String KEYSPACE_NAME =
+      System.getProperty("stargate.health_check.keyspace_name", "data_store_health_check");
+  private static final String TABLE_NAME =
+      System.getProperty("stargate.health_check.table_name", "health_table");
+  private static final int INSERT_TTL_SECONDS = 600;
+  private static final boolean STORAGE_CHECK_ENABLED =
+      Boolean.parseBoolean(System.getProperty("stargate.health_check.data_store.enabled", "true"));
+
+  static final String PK_COLUMN_NAME = "pk";
+  static final String VALUE_COLUMN_NAME = "value";
+  static final Table EXPECTED_TABLE =
+      ImmutableTable.builder()
+          .keyspace(KEYSPACE_NAME)
+          .name(TABLE_NAME)
+          .addColumns(
+              ImmutableColumn.create(
+                  PK_COLUMN_NAME, Column.Kind.PartitionKey, Column.Type.Timeuuid),
+              ImmutableColumn.create(VALUE_COLUMN_NAME, Column.Kind.Regular, Column.Type.Varchar))
+          .build();
+  private final DataStoreFactory dataStoreFactory;
+
+  public StorageHealthChecker(DataStoreFactory dataStoreFactory)
+      throws ExecutionException, InterruptedException {
+
+    this.dataStoreFactory = dataStoreFactory;
+    if (SHOULD_CREATE_KS_AND_TABLE) {
+      ensureTableExists(dataStoreFactory.createInternal());
+    }
+  }
+
+  @Override
+  protected Result check() throws Exception {
+    if (!STORAGE_CHECK_ENABLED) {
```

Comment:
Why not put this in the activator and if it isn't enable then just don't register it? Otherwise we're going to either create the keyspace/table even when it's not enabled or require setting 2 different flags to disable this check.

Reply:
If we do not register a standard checker, we also have to make changes to `BundleService.defaultHealthCheckNames()` to not return its name. I'd be fine with that, but it looks like a cross-cutting concern... As for me, registering, but not running the check is cleaner.